File: 4-Estruturas/Ex3357.java
import java.util.Scanner;

public class Ex3357 { // para ser aceito no URI mude o nome da classe para Main
    public static void main(String[] args) {
        // variaveis e objetos
        Scanner scanner = new Scanner(System.in);

        // entrada
        int n = scanner.nextInt();
        double lGarrafa = scanner.nextDouble();
        double lCuia = scanner.nextDouble();
        scanner.nextLine();
        String entrada = scanner.nextLine();
        String[] nomes = entrada.split("\\s");

        // processamento
        double aux = lGarrafa;
        int cont = -1;
        while (true) {
            cont++;
            if (aux - lCuia <= 0) {
                break;
            } else {
                aux = Math.round((aux - lCuia) * 10.0) / 10.0;
            }
        }

        String nome = nomes[cont % n];

        // saida
        System.out.printf("%s %.1f\n", nome, aux);
        scanner.close();
    }
}

File: 4-Estruturas/ex3357.py
#entrada
n, lGarrafa, lCuia = str(input()).split()
n = int(n)
lGarrafa = float(lGarrafa)
lCuia = float(lCuia)

nomes = str(input()).split()

#processamento
aux = lGarrafa
cont = -1
while True:
    cont += 1
    if aux - lCuia <= 0:
        break
    else:
        aux = round(aux - lCuia, 1)

nome = nomes[cont % n]

#saida
print('{} {:.1f}'.format(nome, aux))
